Assets/HoloToolkit/UX/Scripts/Buttons/ButtonTextProfile.cs

```diff
@@ -0,0 +1,82 @@
+﻿//
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+//
+using UnityEngine;
+
+namespace HoloToolkit.Unity.Buttons
+{
+    public class ButtonTextProfile : ButtonProfile
+    {
+        // Text Mesh settings
+        public TextAlignment Alignment = TextAlignment.Center;
+        public TextAnchor Anchor = TextAnchor.MiddleCenter;
+        public FontStyle Style = FontStyle.Normal;
+        public int Size = 72;
+        public Color Color = Color.white;
+        public Font Font;
+
+        /// <summary>
+        /// Convenience function for getting the offset from an anchor setting
+        /// </summary>
+        /// <param name="anchor"></param>
+        /// <returns></returns>
+        public Vector3 GetOffset (TextAnchor anchor)
+        {
+            Vector3 offset = Vector3.zero;
+            switch (anchor)
+            {
+                case TextAnchor.LowerCenter:
+                    offset = AnchorLowerCenterOffset;
+                    break;
+
+                case TextAnchor.LowerLeft:
+                    offset = AnchorLowerLeftOffset;
+                    break;
+
+                case TextAnchor.LowerRight:
+                    offset = AnchorLowerRightOffset;
+                    break;
+
+                case TextAnchor.MiddleCenter:
+                default:
```

Comment:
default case should always be last and `throw new ArgumentOutOfRangeException("anchor", anchor, null);`